rcsb/utils/insilico3d/ModelReorganizer.py

```diff
@@ -443,6 +447,29 @@ def __addDepositedAssembly(self, dataContainer):
 
         return dataContainer
 
+    def __addNcbiId(self, dataContainer, ncbiId):
+        """Add the NCBI ID to _ma_target_ref_db_details.ncbi_taxonomy_id
+
+        Args:
+            dataContainer (object): mmcif.api.DataContainer object instance
+            ncbiId (str): NCBI ID (e.g., "2779801")
+
+        Returns:
+            dataContainer: updated dataContainer object
+
+        """
+        if not dataContainer.exists("ma_target_ref_db_details"):
+            logger.warning("Category ma_target_ref_db_details does not exist for entry %r", dataContainer.getName())
+            return dataContainer
+
+        logger.debug("Adding NCBI ID for %s", dataContainer.getName())
+        tObj = dataContainer.getObj("ma_target_ref_db_details")
+        if not tObj.hasAttribute("ncbi_taxonomy_id"):
+            tObj.appendAttribute("ncbi_taxonomy_id")
+        tObj.setValue(ncbiId, "ncbi_taxonomy_id", 0)
```

Comment:
Yeah I thought about that too. Correct me if I'm wrong, but only one `ncbi_taxonomy_id` needs to be present for the organism to get populated, right? If so—and given that this is a hack—I didn't seem necessary to spend the extra effort trying to iterate over every row. If you prefer I do though, I can implement that change (or at the very least a comment acknowledging that fact).

Reply:
Another reason for not applying it to every row is based on our observation of the baker models, in which different entities can have different taxonomies. So, again, I went with the simpler solution, since this should only be applied to cases that have no taxonomy whatsoever. We wouldn't want to incidentally assign the wrong taxonomy to an entity from a different organism (if the entry contains chains from different strains of yeast, for example).

Reply:
I see your point and I don't want to make this any more complicated especially since it is hopefully a one time fix. The only reason I was considering this is because the `__addNcbiId` method seems generic but it is actually only applicable to monomers at present. Perhaps add a comment and leave it as is.   

And, yes only one taxonomy id is required for the organism to be populated unless the entities are from different strains like the `ma-bak-cepc` models.

Reply:
OK, I see your point too. Yeah I'll add a comment clarifying that in the method. If we ever encounter a case in the future where we have new models with multiple entities and no taxonomy information (e.g., the select ma-bak-cepc models), we have bigger problems that we'll likely need to address in a different way entirely.